integration-tests/chain-signatures/tests/cases/mod.rs

```diff
@@ -353,3 +355,38 @@ async fn test_multichain_reshare_with_lake_congestion() -> anyhow::Result<()> {
     })
     .await
 }
+
+#[test(tokio::test)]
+async fn test_multichain_update_contract() -> anyhow::Result<()> {
+    let config = MultichainConfig::default();
+    with_multichain_nodes(config.clone(), |ctx| {
+        Box::pin(async move {
+            // Get into running state and produce a singular signature.
+            let state = wait_for::running_mpc(&ctx, Some(0)).await?;
+            wait_for::has_at_least_mine_triples(&ctx, 2).await?;
+            wait_for::has_at_least_mine_presignatures(&ctx, 1).await?;
+            actions::single_payload_signature_production(&ctx, &state).await?;
+
+            // Perform update to the contract and see that the nodes are still properly running and picking
+            // up the new contract by first upgrading the contract, then trying to generate a new signature.
+            let id = ctx.propose_update_contract_default().await;
+            ctx.vote_update(id).await;
+            tokio::time::sleep(std::time::Duration::from_secs(3)).await;
+            wait_for::has_at_least_mine_presignatures(&ctx, 1).await?;
+            actions::single_payload_signature_production(&ctx, &state).await?;
+
+            // Now do a config update and see if that also updates the same:
+            let id = ctx.propose_update(ProposeUpdateArgs {
```

Comment:
Same with config. We should propose something new, then fetch the config and compere it to what we proposed.

Reply:
Same thing I said in the previous comment, but we should see if introducing interesting configs would mess up the nodes or not like high concurrent introduction and concurrent generation